exampleCourse/questions/examplesRGetData/server.py

```diff
@@ -0,0 +1,62 @@
+import io
+import numpy
+import random
+import rpy2.robjects as robjects
+from pandas import DataFrame
+
+def generate(data):
+    
+    # Get list of data sets in base R
+    # val = robjects.r('d = data(package = "datasets")$results[,"Item"]; d[!grepl("[[:space:]]", d)]')
+    
+    r_datasets = ['CO2', 'attitude', 'USJudgeRatings', 'pressure', 'warpbreaks', 'esoph', 'ToothGrowth']
+    
+    # Pick data set
+    ds_id = random.randint(0, len(r_datasets) - 1)
+    
+    # Get the Data Set Name
+    ds_name = r_datasets[ds_id]
+    
+    # Find dimensions
+    ds_dim = robjects.r("dim(get('%s'))" % (ds_name))
+    
+    # Retrieve individual dimensions
+    ds_nrow = ds_dim[0]
+    ds_ncol = ds_dim[1]
+    
+    if ds_nrow > 15:
+       ds_nrow = random.randint(6, 15)
+       
+    # Simulate a location
+    #loc_row = random.randint(1, ds_nrow)
+    #loc_col = random.randint(1, ds_ncol)
+    
+    # Retrieve location value
+    #loc_value = robjects.r("get('%s')[%s, %s]" % (ds_name, loc_row, loc_col))
+    #data['correct_answers']['ds_loc_val'] = loc_value
+    
+    # Output Data Table
+    data['params']['display_data'] = str(robjects.r("knitr::kable(head(get('%s'), n = %s), align = rep('c', %s), format = 'html', row.names = FALSE)" % 
```

Comment:
Would the R code be easier to read with f-strings? Like this:
```suggestion
    data['params']['display_data'] = str(robjects.r(f"knitr::kable(head(get('{dn_name}'), n = {ds_nrow}), align = rep('c', {ds_ncol}), format = 'html', row.names = FALSE)"))
```